pandas/core/generic.py

```diff
@@ -4957,6 +4958,8 @@ def sample(
             n = 1
         elif n is not None and frac is None and n % 1 != 0:
             raise ValueError("Only integers accepted as `n` values")
+        elif n is None and frac is not None and frac <= 0:
+            raise ValueError("Only positive floats accepted as `frac` values")
```

Comment:
I don't think we actually need to explicitly write this check since, as shown in the issue, this validation already occurs after an `n` is computed from `frac` (conversion in the `elif` below this one, negative check a few lines below that).

It is interesting that this didn't cause the test suite to fail, as we check the messages associated with the errors.   That probably means we don't have a test for the `frac < 0` case, so would be nice to add one.